File: src/main/java/com/lmp/shared/config/site/SiteConfigEntry.java
package com.lmp.shared.config.site;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;

@Entity
@Table(name = "site_config")
public class SiteConfigEntry {

    @Id
    @Column(name = "config_key", length = 128, nullable = false)
    private String key;

    @Column(name = "config_value", length = 4000)
    private String value;

    @Column(name = "config_description", length = 500)
    private String description;

    @UpdateTimestamp
    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt;

    protected SiteConfigEntry() {}

    public SiteConfigEntry(String key, String value, String description) {
        this.key = key;
        this.value = value;
        this.description = description;
    }

    public String getKey() { return key; }
    public void setKey(String key) { this.key = key; }

    public String getValue() { return value; }
    public void setValue(String value) { this.value = value; }

    public String getDescription() { return description; }
    public void setDescription(String description) { this.description = description; }

    public LocalDateTime getUpdatedAt() { return updatedAt; }
}

File: backend-dotnet/src/Lmp.Domain/SiteConfiguration/SiteConfigEntry.cs
namespace Lmp.Domain.SiteConfiguration;

/// <summary>
/// A hot-editable site configuration entry — maps to <c>site_config</c>
/// (port of <c>com.lmp.shared.config.site.SiteConfigEntry</c>).
/// </summary>
public class SiteConfigEntry
{
    public string Key { get; set; } = null!;
    public string? Value { get; set; }
    public string? Description { get; set; }
    public DateTime UpdatedAt { get; set; }

    public SiteConfigEntry()
    {
    }

    public SiteConfigEntry(string key, string? value, string? description)
    {
        Key = key;
        Value = value;
        Description = description;
    }
}
